lib/spack/spack/util/environment.py

```diff
@@ -72,4 +72,4 @@ def dump_environment(path):
     """Dump the current environment out to a file."""
     with open(path, 'w') as env_file:
         for key, val in sorted(os.environ.items()):
-            env_file.write("%s=%s\n" % (key, val))
+            env_file.write("export %s=\"%s\"\n" % (key, val))
```

Comment:
Can you use `'export %s="%s"\n'` instead? Backslashes are evil.

Reply:
absolutely.  pushed new version doing things that way.